pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -10,6 +10,10 @@
 , zlib ? null, extraPackages ? [], extraBuildCommands ? ""
 , dyld ? null # TODO: should this be a setup-hook on dyld?
 , isGNU ? false, isClang ? cc.isClang or false, gnugrep ? null
+, # Prefix for binaries. Customarily ends with a dash separator.
+  prefix ? stdenv.lib.optionalString (targetPlatform != hostPlatform)
+                                     (targetPlatform.config + "-")
```

Comment:
It used to not be a parameter, but I figured technically prefixing is a matter of taste, and users might want things to not be prefixed.

But it probably is a bad idea is the underlying compiler will already be prefixed or not, and as you noticed the llvm stuff isn't prefixed so it's conditional too.

So either I can just let-bind it, or I'll have a separate "in prefix" and "out prefix", and define is prefix to be non-`""` only when cross compiling *and* using GCC. Your call :).